app/vmui/packages/vmui/src/components/CustomPanel/Views/GraphView.tsx

```diff
@@ -33,7 +34,7 @@ const promValueToNumber = (s: string): number => {
     case "-Inf":
       return -Infinity;
     default:
-      return parseFloat(s);
+      return parseFloat(`${whole}.${decimal.slice(0, 10)}`);
```

Comment:
This fix breaks if the request returns small values near zero. For example, the request `rand()*1e-20` would draw zero line, while it is expected that it draws random values line in the range `[0 ... 1e-20]`.

The original issue is in [uPlot library](https://github.com/leeoniya/uPlot) - it stops working when drawing time series with small difference between time series values comparing to the average series value. E.g. if `abs(avg(values)) > range(values) * 1e10` . For example, it freezes when drawing the response for `1e8 + rand()*1e-7` [MetricsQL query](https://docs.victoriametrics.com/MetricsQL.html).

The solution is to report the bug to the upstream [uPlot repository](https://github.com/leeoniya/uPlot) and fix the bug there. In the meantime vmui can provide a workaround, which would replace time series values with their average value before passing them to uPlot if `abs(avg(values)) > range(values)*1e10`.